### ISSUE TITLE
Connecting in monitor-mode results in timeout

### ISSUE THREAD

**Issue by chill-nemesis** [Mecademic/mecademicpy#4]
### Expected behaviour
Calling `robot.Connect(address, monitor_mode=True)` connects to the robot on the monitoring port only.

### Actual behaviour
Calling `robot.Connect(address, monitor_mode=True)` raises a `CommunicationError` (reason: Timeout)

### Minimally crashing example
``` python
import mecademicpy as mr

robotAddress='192.168.0.100'

with mr.Robot() as robot:
    robot.Connect(robotAddress, monitor_mode=True)
```

### Additional information
 - python v.3.10.2
 - mecademicpy v.1.1.0
 - Meca500 R3 v8.3.10

### Method-call trace:
 - `robot.Connect()`
 - `robot._initialize_monitoring_connection()`
 - `robot._receive_welcome_message()`

When initiating a monitor-mode only connection via `robot.Connect(address, monitor_mode=True)`, the method raises a `CommunicationError`.
 This can be traced back to the `_receive_welcom_message(...)` method in robot.py to the check of
``` python
if response.id != mx_def.MX_ST_CONNECTED:
    self.logger.error('Connection error: {}'.format(response))
    raise CommunicationError('Connection error: {}'.format(response))
```
To my knowledge, when initializing a monitor-connection, your code does expect expect a welcome message on the monitor-port from the robot, which is not sent (at least I did not receive one even after enabling debug-logging - only the expected monitor messages as described in the programmer manual. If necessary, I can provide a recording of the received monitor messages). Because no such message could be received, the while-loop in `_receive_welcome_message` eventually runs into the timeout check, which then triggers the above listed condition to raise a CommunicationError.
Unfortunately, just ignoring the welcome message is not a viable option, since parts of your framework rely upon the correctly set `robot._robot_info` member from the welcome message. However, I am not sure if you can reasonably expect a welcome message when you establish a monitor connection - at least, nothing the like is mentioned in the meca500 manuals.

My current solution to circumvent this problem is to detect the monitor connection (indicated by the `from_command_port == False` parameter in `_receive_welcome_message`) and simulate receiving the expected robot message string:

``` python
def _receive_welcome_message(self, message_queue: queue.Queue, from_command_port: bool):
        """Receive and parse a welcome message in order to set _robot_info and _robot_rt_data.

        Parameters
        ----------
        message_queue : queue
            The welcome message will be fetched from this queue.
        """
        response = _Message(None, None)

        # HACK: we are in monitor mode. The driver expects a specific message which is not sent by the robot
        # (for some reason)
        # We thus fake the message (for now) for our meca model
        if not from_command_port:
            self.logger.warn("Emulating meca robot version!")
            response = _Message.from_string("[3000][Connected to Meca500 R3 v8.3.10]")

        start = time.time()
        while response.id != mx_def.MX_ST_CONNECTED:
            try:
```

**Comment by chill-nemesis** [comment]
I have updated the robot to the latest available version (8.4.4), which resolves the issue.

However, since the driver is supposed to also support Firmware 8.3 I'll keep this issue open.

**Comment by gbmecademic** [comment]
Hello,
I will notify the dev team so that they can fix this.

**Comment by gbmecademic** [comment]
This issue was fixed in version 1.2.1. I am closing this issue.
